packages/astro/src/check.ts

```diff
@@ -0,0 +1,115 @@
+import { AstroCheck, DiagnosticSeverity } from '@astrojs/language-server';
+import type { AstroConfig } from './@types/astro';
+import { bold, blue, black, bgWhite, red, cyan, yellow } from 'kleur/colors';
+import glob from 'fast-glob';
+import * as path from 'path';
+import { pathToFileURL } from 'url';
+import * as fs from 'fs';
+
+async function openAllDocuments(
+  workspaceUri: URL,
+  filePathsToIgnore: string[],
+  checker: AstroCheck
+) {
+  const files = await glob('**/*.astro', {
+      cwd: workspaceUri.pathname,
+      ignore: ['node_modules/**'].concat(filePathsToIgnore.map((ignore) => `${ignore}/**`))
+  });
+  const absFilePaths = files.map((f) => path.resolve(workspaceUri.pathname, f));
+
+  for (const absFilePath of absFilePaths) {
+      const text = fs.readFileSync(absFilePath, 'utf-8');
+      checker.upsertDocument(
+          {
+              uri: pathToFileURL(absFilePath).toString(),
+              text
+          }
+      );
+  }
+}
+
+interface Result {
+  errors: number;
+  warnings: number;
+}
+
+function offsetAt({ line, character }: { line: number, character: number; }, text: string) {
+  let i = 0;
+  let l = 0;
+  let c = 0;
+  while(i < text.length) {
+    if(l === line && c === character) {
+      break;
+    }
+
+    let char = text[i];
+    switch(char) {
+      case '\n': {
+        l++;
+        c = 0;
+        break;
+      }
+      default: {
+        c++;
+        break;
+      }
+    }
+
+    i++;
+  }
+
+  return i;
+}
+
+function pad(str: string, len: number) {
+  return Array.from({ length: len }, () => str).join('');
+}
+
+export async function run() {
+
+}
+
+export async function check(astroConfig: AstroConfig) {
+  const root = astroConfig.projectRoot;
+  let checker = new AstroCheck(root.toString());
+  await openAllDocuments(root, [], checker);
+
+  let diagnostics = await checker.getDiagnostics();
+
+  let result: Result = {
+    errors: 0,
+    warnings: 0
+  };
+
+  diagnostics.forEach(diag => {
+    diag.diagnostics.forEach(d => {
+      switch(d.severity) {
+        case DiagnosticSeverity.Error: {
+          console.error(`${bold(cyan(path.relative(root.pathname, diag.filePath)))}:${bold(yellow(d.range.start.line))}:${bold(yellow(d.range.start.character))} - ${d.message}`);
```

Comment:
can you add a `eslint-disable no-console` comment somewhere on this page? See packages/astro/src/cli/index.ts for an example

Reply:
Done, thanks!